pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -487,8 +463,19 @@ func updateCheckState(checkState *kueue.AdmissionCheckState, state kueue.CheckSt
 	return true
 }
 
-func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, checks []string, activeOrLastPRForChecks map[string]*autoscaling.ProvisioningRequest) error {
+func updateWorkloadInfo(wl *kueue.Workload, wlInfo *workloadInfo) {
```

Comment:
nit: let's make this a function received by `workloadInfo` so we can use `workloadInfo.update(wl)`

Reply:
Done